From: Michael Olbrich <m.olbrich@pengutronix.de>
Subject: remove make from the run_tests script

Using make in the run_tests script only works in the build tree or we would
have to install a lot of useless stuff. This patch adds the necessary stuff to
run_tests to avoid calling make.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>

---
 testcases/open_posix_testsuite/run_tests |   56 ++++++++++++++++++++++++++++---
 1 file changed, 52 insertions(+), 4 deletions(-)

Index: b/testcases/open_posix_testsuite/run_tests
===================================================================
--- a/testcases/open_posix_testsuite/run_tests
+++ b/testcases/open_posix_testsuite/run_tests
@@ -20,11 +20,59 @@ in the POSIX spec
 EOF
 }
 
+LOGFILE=/tmp/posix-test-logfile
+TIMEOUT_RET=`./t0 0; echo $?`
+
+runtest()
+{
+	COMPLOG=$LOGFILE.$$
+	[ -f $1 ] || exit 0
+	$TIMEOUT $1 > $COMPLOG 2>&1
+	RESULT=$?
+	if [ $RESULT -eq 1 ]
+	then
+		MSG="FAILED"
+	fi
+	if [ $RESULT -eq 2 ]
+	then
+		MSG="UNRESOLVED"
+	fi
+	if [ $RESULT -eq 4 ]
+	then
+		MSG="UNSUPPORTED"
+	fi
+	if [ $RESULT -eq 5 ]
+	then
+		MSG="UNTESTED"
+	fi
+	if [ $RESULT -eq $TIMEOUT_RET ]
+	then
+		MSG="HUNG"
+	fi
+	if [  $RESULT -gt 5  -a  $RESULT -ne $TIMEOUT_RET  ]
+	then
+		MSG="INTERRUPTED"
+	fi
+	if [ $RESULT -eq 0 ]
+	then
+		echo "$1: execution: PASS" | tee -a $LOGFILE
+	else
+		(
+			echo "$1: execution: $MSG: Output: "
+			cat $COMPLOG
+		) >> $LOGFILE
+		echo "$1: execution: $MSG "
+	fi
+	rm -f $COMPLOG
+}
+
 runtests()
 {
 	for test in `ls -d $1`; do
-		POSIX_TARGET=$test make build-tests
-		POSIX_TARGET=$test make run-tests
+		tests=`find $test -name "*.test"`
+		for t in $tests; do
+			runtest $t
+		done
 	done
 }
 
@@ -64,8 +112,8 @@ case $1 in
 	runtests "$BASEDIR/m*map"
 	runtests "$BASEDIR/shm_*"
 	;;
-  *)	usage
-	exit 1
+  *) echo "Executing ALL tests"
+  	runtests .
 	;;
 esac
 
